src/main/resources/global/90-init-alfresco.sh

```diff
@@ -26,9 +26,20 @@ fi
 if [ -n "$CATALINA_HOME" ]; then
 
   user="tomcat"
-  if [[ $(stat -c %U /opt/alfresco/alf_data) != "$user" ]]; then
-    chown -R $user:$user /opt/alfresco/alf_data
+  # Check existence of alf_data/contentstore
+  # -> if dir exists, check if the permissions are incorrect
+  # --> if permissions are incorrect, exit with error code
+  # -> else if not exists, chown only alf_data
+  if [[ -e /opt/alfresco/alf_data/contentstore \
+   && ($(stat -c %U /opt/alfresco/alf_data) != "$user" && $(stat -c %a /opt/alfresco/alf_data) -lt 766) ]]; then
+     # custom exit code for debug to this script
+     echo "Contentstore exists within /opt/alfresco/alf_data , but directory does not have correct ownership/permission to run alfresco. Exiting with code 64."
+     exit 64
+  else
+    echo "No contentstore in alf_data, chowning directory to tomcat user."
```

Comment:
As I proposed yesterday, it's important that people understand the **why** and this is more the **what**. Suggestion:
echo "No contentstore found in alf_data -> Assuming development scenario -> Chowning directory to tomcat user."

Reply:
The comment at l36 for instance, does explain it well.